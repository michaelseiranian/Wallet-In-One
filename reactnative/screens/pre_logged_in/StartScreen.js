import React from 'react';
import { StatusBar } from 'expo-status-bar';
import {
  StyleSheet,
  Image,
  View,
  Text,
  ImageBackground,
  TouchableOpacity
} from 'react-native';

import { useIsFocused } from '@react-navigation/native';
function FocusAwareStatusBar(props) {
  const isFocused = useIsFocused();
  return isFocused ? <StatusBar {...props} /> : null;
}
import { useTheme } from 'reactnative/src/theme/ThemeProvider';

export default function StartScreen ({ navigation }) {

  const {dark, colors, setScheme} = useTheme();

  const styles = StyleSheet.create({
    button:{
      margin:10,
    },
    background: {
      width: '100%',
      height: '100%'
    },
    logo:{
      width: 280,
      height: 280,
      marginLeft: '15%',
      marginTop: '-15%',
    },
    signup: {
      backgroundColor: colors.primary,
      color: 'white',
      width: "75%",
      borderRadius: 25,
      textAlign: 'center',
      fontWeight: 'bold',
      marginLeft: '11%',
      padding: "2%",
      fontSize:  27,
      marginTop: '65%'
    },
    login: {
      backgroundColor: 'white',
      color: colors.primary,
      width: "75%",
      borderRadius: 25,
      textAlign: 'center',
      fontWeight: 'bold',
      marginLeft: '11%',
      padding: "2%",
      fontSize:  27,
      marginTop: '5%',
    },
    aboutUs: {
      backgroundColor: 'black',
      color: colors.primary,
      width: "30%",
      borderRadius: 25,
      textAlign: 'center',
      fontWeight: 'bold',
      padding: "2%",
      fontSize:  17,
      marginTop: '-10%',
      alignSelf: 'center'
    }
  });
  

    return (
      <ImageBackground
          source={require('reactnative/assets/background.jpg')}
          style={styles.background}
      >
        <View>
          <FocusAwareStatusBar />
          <Image
            source={require('reactnative/assets/logo.png')}
            style={styles.logo}
            resizeMode="contain"
          >
          </Image>
          {/* <StatusBar style="auto" /> */}
          <TouchableOpacity
            onPress={() => navigation.navigate('About Us')}
          >
            <Text style={styles.aboutUs}>About Us</Text>
          </TouchableOpacity>

          <TouchableOpacity
            onPress={() => navigation.navigate('Sign Up')}
          >
            <Text style={styles.signup}>Sign Up</Text>
          </TouchableOpacity>

          <TouchableOpacity
            onPress={() => navigation.navigate('Login')}
          >
            <Text style={styles.login}>Log In</Text>
          </TouchableOpacity>
        </View>
      </ImageBackground>
    );
}