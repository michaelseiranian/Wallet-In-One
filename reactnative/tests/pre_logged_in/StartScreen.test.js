import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import StartScreen from '../../screens/pre_logged_in/StartScreen';
import renderer from 'react-test-renderer';

describe('<StartScreen />', () => {

    it('about us button test', () => {
        const navigate = jest.fn();
        const { getByText } = render(<StartScreen navigation={{ navigate }} />);
        fireEvent.press(getByText('About Us'));
        expect(navigate).toHaveBeenCalledWith('About Us');
    });

    it('sign up button test', () => {
        const navigate = jest.fn();
        const { getByText } = render(<StartScreen navigation={{ navigate }} />);
        fireEvent.press(getByText('Sign Up'));
        expect(navigate).toHaveBeenCalledWith('Sign Up');
    });

    it('login button test', () => {
        const navigate = jest.fn();
        const { getByText } = render(<StartScreen navigation={{ navigate }} />);
        fireEvent.press(getByText('Log In'));
        expect(navigate).toHaveBeenCalledWith('Login');
    });

    it('snapshot startScreen test', () => {
        const navigate = jest.fn();
        const startScreen = renderer.create(<StartScreen navigation={{ navigate }} />);
        expect(startScreen).toMatchSnapshot();         
      });
});