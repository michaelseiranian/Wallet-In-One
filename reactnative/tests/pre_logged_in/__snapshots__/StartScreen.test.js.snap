// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartScreen /> snapshot startScreen test 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View>
    <Image
      resizeMode="contain"
      source={1}
      style={
        {
          "height": 280,
          "marginLeft": "15%",
          "marginTop": "-15%",
          "width": 280,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "black",
            "borderRadius": 25,
            "color": "purple",
            "fontSize": 17,
            "fontWeight": "bold",
            "marginTop": "-10%",
            "padding": "2%",
            "textAlign": "center",
            "width": "30%",
          }
        }
      >
        About Us
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": "purple",
            "borderRadius": 25,
            "color": "white",
            "fontSize": 27,
            "fontWeight": "bold",
            "marginLeft": "11%",
            "marginTop": "65%",
            "padding": "2%",
            "textAlign": "center",
            "width": "75%",
          }
        }
      >
        Sign Up
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": "white",
            "borderRadius": 25,
            "color": "purple",
            "fontSize": 27,
            "fontWeight": "bold",
            "marginLeft": "11%",
            "marginTop": "5%",
            "padding": "2%",
            "textAlign": "center",
            "width": "75%",
          }
        }
      >
        Log In
      </Text>
    </View>
  </View>
</View>
`;
